AWSClientRuntime/Sources/Regions/BundleRegionProvider.swift

```diff
@@ -0,0 +1,40 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+import AwsCommonRuntimeKit
+import ClientRuntime
+import Foundation.NSBundle
+
+public struct BundleRegionProvider: RegionProvider {
+    private let logger: SwiftLogger
+    private let bundle: Foundation.Bundle
+    private let regionKey: String
+    
+    public init(bundle: Foundation.Bundle = Bundle.main, regionKey: String = "AWS_REGION") {
+        self.logger = SwiftLogger(label: "BundleRegionProvider")
+        self.bundle = bundle
+        self.regionKey = regionKey
+    }
+    
+    public func resolveRegion() -> Future<String?> {
+        let future = Future<String?>()
+        #if os(iOS) || os(watchOS) || os(tvOS)
+        future.fulfill(region())
```

Comment:
From a security perspective, if we allow customers to enter "anything" into a plist -- this can be bad -- we probably want to, at the very minimum, limit the size of string we can read from a plist -- e.g. 20 bytes maximum.

Reply:
how can we do that?

Reply:
Done. used `prefix()`